packages/lib/link/messages.js

```diff
@@ -315,6 +316,9 @@ messages.listInstances = new Request({
 					"name": { type: "string" },
 					"id": { type: "integer" },
 					"assigned_slave": { type: ["null", "integer"] },
+					"assigned_slave_name": { type: ["null", "string"] },
```

Comment:
Currently slave IDs are mapped to slave name independently in the places where it is required. Always providing it with the request is useful and does cut down on one request in many cases while consuming minimal bandwidth, but it is still a calculated parameter. @Hornwitser probably has some thoughts on this. Also, if we are doing it this way we should probably use this field on the instance view page instead of using an additional request.

Reply:
The IDs are resolved to names on the control client by querying the slave list and looking it up. The reasoning is that I eventually want to support pushing instance and slave status updates to the  control client (which includes name changes) so it would be easier convert the system to this if the control client did the ID to name translation itself.

Reply:
`assigned_slave` is used in the InstanceViewPage so you can't just remove it from here without breaking something.